src/disk_io_thread.cpp

```diff
@@ -1685,52 +1708,48 @@ namespace libtorrent
-			&& m_settings.get_bool(settings_pack::use_write_cache))
-		{
-			TORRENT_ASSERT((r.start % m_disk_cache.block_size()) == 0);
-			j->flags |= disk_io_job::use_disk_cache;
 
-			if (storage->is_blocked(j))
-			{
-				// this means the job was queued up inside storage
-				m_stats_counters.inc_stats_counter(counters::blocked_disk_jobs);
-				DLOG("blocked job: %s (torrent: %d total: %d)\n"
-					, job_action_name[j->action], j->storage ? j->storage->num_blocked() : 0
-					, int(m_stats_counters[counters::blocked_disk_jobs]));
-				// make the holder give up ownership of the buffer
-				// since the job was successfully queued up
-				buffer.release();
-				return;
-			}
+		TORRENT_ASSERT((r.start % m_disk_cache.block_size()) == 0);
 
-			mutex::scoped_lock l(m_cache_mutex);
-			// if we succeed in adding the block to the cache, the job will
-			// be added along with it. we may not free j if so
-			cached_piece_entry* dpe = m_disk_cache.add_dirty_block(j);
+		if (storage->is_blocked(j))
+		{
+			// this means the job was queued up inside storage
+			m_stats_counters.inc_stats_counter(counters::blocked_disk_jobs);
+			DLOG("blocked job: %s (torrent: %d total: %d)\n"
+				, job_action_name[j->action], j->storage ? j->storage->num_blocked() : 0
+				, int(m_stats_counters[counters::blocked_disk_jobs]));
+			// make the holder give up ownership of the buffer
+			// since the job was successfully queued up
+			buffer.release();
+			return;
+		}
 
-			// if the buffer was successfully added to the cache
-			// our holder should no longer own it
-			if (dpe) buffer.release();
+		mutex::scoped_lock l(m_cache_mutex);
+		// if we succeed in adding the block to the cache, the job will
+		// be added along with it. we may not free j if so
+		cached_piece_entry* dpe = m_disk_cache.add_dirty_block(j);
 
-			if (dpe && dpe->outstanding_flush == 0)
-			{
-				dpe->outstanding_flush = 1;
-				l.unlock();
+		// if the buffer was successfully added to the cache
+		// our holder should no longer own it
+		if (dpe) buffer.release();
 
-				// the block and write job were successfully inserted
-				// into the cache. Now, see if we should trigger a flush
-				j = allocate_job(disk_io_job::flush_hashed);
-				j->storage = storage->shared_from_this();
-				j->piece = r.piece;
-				j->flags = flags;
-				add_job(j);
-			}
-			// if we added the block (regardless of whether we also
-			// issued a flush job or not), we're done.
-			if (dpe) return;
+		if (dpe && dpe->outstanding_flush == 0)
+		{
+			dpe->outstanding_flush = 1;
 			l.unlock();
-		}
+
+			// the block and write job were successfully inserted
+			// into the cache. Now, see if we should trigger a flush
+			j = allocate_job(disk_io_job::flush_hashed);
+			j->storage = storage->shared_from_this();
+			j->piece = r.piece;
+			j->flags = flags;
+			add_job(j);
+		}
+		// if we added the block (regardless of whether we also
+		// issued a flush job or not), we're done.
+		if (dpe) return;
+		l.unlock();
```

Comment:
Yeah, this is all just reducing the indent after removing the write cache enabled check.